app/controllers/user_api_keys_controller.rb

```diff
@@ -62,7 +62,11 @@ def create
     scopes = params[:scopes].split(",")
 
     # destroy any old keys we had
-    UserApiKey.where(user_id: current_user.id, client_id: params[:client_id]).destroy_all
+    UserApiKey.where(
+      "user_id = :user_id OR client_id = :client_id",
```

Comment:
Does this risk deleting the user's keys for other DiscourseHub installations (on other devices, for example)? I think you're right, there's an issue to fix here, but I'm not sure if this goes a bit too far.

Reply:
Maybe we do `UserApiKey.where(client_id: params[:client_id]).destroy_all`?